tests/components/test_proximity.py

```diff
@@ -7,21 +7,89 @@
 
 import homeassistant.core as ha
 from homeassistant.components import proximity
+from homeassistant.components import zone
+import homeassistant.components.device_tracker as device_tracker
+import homeassistant.util.dt as dt_util
+from datetime import datetime, timedelta
 
 class TestProximity:
     """ Test the Proximity component. """
 
     def setup_method(self, method):
         self.hass = ha.HomeAssistant()
```

Comment:
Since you're writing to known_devices.yaml you have to use a test Home Assistant so you do not mess up the users directory.

```python
from tests.common import get_test_home_assistant

self.hass = get_test_home_assistant()
```